rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -518,9 +518,12 @@ RecorderImpl::create_subscription(
     topic_name,
     topic_type,
     qos,
-    [this, topic_name, topic_type](std::shared_ptr<const rclcpp::SerializedMessage> message) {
+    [this, topic_name, topic_type](std::shared_ptr<const rclcpp::SerializedMessage> message,
+    const rclcpp::MessageInfo & mi) {
       if (!paused_.load()) {
-        writer_->write(message, topic_name, topic_type, node->get_clock()->now());
+        writer_->write(
+          message, topic_name, topic_type,
+          rclcpp::Time(mi.get_rmw_message_info().received_timestamp));
```

Comment:
this can be better system time stamp in the rosbag2, but one question.

what if `--use-sim-time` option is enabled with `ros2 bag record`? i guess it does not work as expected? rmw is not aware of ros simulation time clock, so this will always store the system time from rmw implementation.

Reply:
fair point, i will rework the patch for this case